cmd/tmsUpload.go

```diff
@@ -0,0 +1,288 @@
+package cmd
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/url"
+	"sort"
+	"strconv"
+
+	"github.com/SAP/jenkins-library/pkg/command"
+	piperHttp "github.com/SAP/jenkins-library/pkg/http"
+	"github.com/SAP/jenkins-library/pkg/log"
+	"github.com/SAP/jenkins-library/pkg/piperutils"
+	"github.com/SAP/jenkins-library/pkg/telemetry"
+	"github.com/SAP/jenkins-library/pkg/tms"
+	"github.com/ghodss/yaml"
+	"github.com/pkg/errors"
+)
+
+type uaa struct {
+	Url          string `json:"url"`
+	ClientId     string `json:"clientid"`
+	ClientSecret string `json:"clientsecret"`
+}
+
+type serviceKey struct {
+	Uaa uaa    `json:"uaa"`
+	Uri string `json:"uri"`
+}
+
+type tmsUploadUtils interface {
+	command.ExecRunner
+
+	FileExists(filename string) (bool, error)
+	FileRead(path string) ([]byte, error)
+
+	// Add more methods here, or embed additional interfaces, or remove/replace as required.
+	// The tmsUploadUtils interface should be descriptive of your runtime dependencies,
+	// i.e. include everything you need to be able to mock in tests.
+	// Unit tests shall be executable in parallel (not depend on global state), and don't (re-)test dependencies.
+}
+
+type tmsUploadUtilsBundle struct {
+	*command.Command
+	*piperutils.Files
+
+	// Embed more structs as necessary to implement methods or interfaces you add to tmsUploadUtils.
+	// Structs embedded in this way must each have a unique set of methods attached.
+	// If there is no struct which implements the method you need, attach the method to
+	// tmsUploadUtilsBundle and forward to the implementation of the dependency.
+}
+
+func newTmsUploadUtils() tmsUploadUtils {
+	utils := tmsUploadUtilsBundle{
+		Command: &command.Command{},
+		Files:   &piperutils.Files{},
+	}
+	// Reroute command output to logging framework
+	utils.Stdout(log.Writer())
+	utils.Stderr(log.Writer())
+	return &utils
+}
+
+func tmsUpload(config tmsUploadOptions, telemetryData *telemetry.CustomData, influx *tmsUploadInflux) {
+	// Utils can be used wherever the command.ExecRunner interface is expected.
+	// It can also be used for example as a mavenExecRunner.
+	utils := newTmsUploadUtils()
+	client := &piperHttp.Client{}
+	proxy := config.Proxy
+	if proxy != "" {
+		transportProxy, err := url.Parse(proxy)
+		if err != nil {
+			log.Entry().WithError(err).Fatalf("Failed to parse proxy string %v into a URL structure", proxy)
```

Comment:
does call to log.Entry().WithError(err).Fatalf() or log.Entry().WithError(err).Fatal() result in panic ? I see our code assumes it does.